frontend/map.component.ts

```diff
@@ -1,23 +1,59 @@
 import { Component, OnInit } from "@angular/core";
 
-import { Tweet } from "./tweet";
 import { MapService } from "./map.service";
+import { MapObject } from "./mapObject";
+import { Tweet } from "./tweet";
 
 @Component({
     selector: "map-app",
     templateUrl: "map.component.html",
     providers: [MapService]
 })
 export class MapComponent implements OnInit {
+
+    map: google.maps.Map;
+    currentlyOpenInfoWindow: google.maps.InfoWindow = null;
     tweets: Tweet[];
 
     constructor(private mapService: MapService) {}
 
+    closeCurrentlyOpenInfoWindow() {
+        if (this.currentlyOpenInfoWindow !== null) {
+            this.currentlyOpenInfoWindow.close();
+        }
+    }
+
+    drawMapObject(mapObject: MapObject) {
+        const latLng = new google.maps.LatLng(mapObject.locationLat, mapObject.locationLong);
+        const infoWindow = new google.maps.InfoWindow({
+            content: mapObject.name
+        });
+        const marker = new google.maps.Marker({
+            position: latLng,
+            title: mapObject.name
+        });
+
+        marker.addListener("click", (() => {
+            this.closeCurrentlyOpenInfoWindow();
+            this.showInfoWindowForMarker(marker, infoWindow);
+        }));
+        marker.setMap(this.map);
+    }
+
+    drawMapObjects(mapObjects: MapObject[]) {
+        mapObjects.map((mapObject) => this.drawMapObject(mapObject));
+    }
+
+    getMapObjects() {
+        this.mapService.getMapObjects().then(mapObjects => this.drawMapObjects(mapObjects));
+    }
+
     getTweets() {
         this.mapService.getTweets().then(tweets => this.tweets = tweets);
     }
 
     ngOnInit() {
+        this.getMapObjects();
```

Comment:
Das sollte erst nach Zeile 67 aufgerufen werden, da im Callback `this.map` benutzt wird.